src/main/scala/viper/gobra/Gobra.scala

```diff
@@ -42,38 +42,88 @@ object GoVerifier {
   }
 }
 
-trait GoVerifier {
+trait GoVerifier extends StrictLogging {
+
+  type Warning = String
 
   def name: String = {
     this.getClass.getSimpleName
   }
 
-  def verify(config: Config)(executor: GobraExecutionContext): Future[VerifierResult] = {
-    verify(config.inputs, config)(executor)
+  def verifyAllPackages(config: Config)(executor: GobraExecutionContext): Vector[VerifierError] = {
+    val statsCollector = StatsCollector(config.reporter)
+    var warningCount: Int = 0
+    var allErrors: Vector[VerifierError] = Vector()
+
+    // write report to file on shutdown
+    Runtime.getRuntime.addShutdownHook(new Thread() {
+      override def run(): Unit = {
+        val statsFile = config.gobraDirectory.resolve("stats.json").toFile
+        logger.info("Writing report to " + statsFile.getPath)
+        statsCollector.writeJsonReportToFile(statsFile)
+      }
+    })
+
+    config.inputPackageMap.foreach({ case (pkg, inputs) =>
+      val taskName = pkg.path + " - " + pkg.name
```

Comment:
I don't think if I have understood the purpose of `taskName`. It's quite cumbersome to pass this around throughout Gobra but it's still an internal identifier that is meaningless to Gobra users. Can't we derive a unique identifier when we need it (e.g. by having some getter function in an object, on Config or on an AST node)

Reply:
I put the `taskName` inside of the `Config` to avoid passing it around